var/spack/repos/builtin/packages/cp2k/package.py

```diff
@@ -200,7 +200,7 @@ class Cp2k(MakefilePackage, CMakePackage, CudaPackage, ROCmPackage):
         depends_on("libxc@5.1.3:5.1", when="@8.2:8")
         depends_on("libxc@5.1.7:5.1", when="@9:2022.2")
         depends_on("libxc@6.1:", when="@2023.1:")
-        depends_on("libxc@6.2:", when="@2023.2:")
+        depends_on("libxc@6.2:6.2.2", when="@2023.2:2024.3")
```

Comment:
If I'm reading this correctly, you would now allow `cp2k@master` to be concretized with `libxc@6.1`, which was not previously possible and I think not intended? The `depends_on` that was modified used to cover `master` (and e.g. a future `2024.4` or newer) but now no longer covers it. Do you maybe want to add another `depends_on("libxc@6.2:, when="@2024.4:")` or `when="@master"`? Alternatively, you could keep the old `depends_on`, but only add a `conflicts("libxc@6.2.3:", when="@2023.2:2024.3")` instead of the updated `depends_on`.

Reply:
From the title it sounds like a forward compat bound needs to be added. That's usually done as a separate statement:

```python
depends_on("libxc@:6", when="@:2024.3")
```

it means libxc 7 support was only added after 2024.3

Reply:
Thanks for the advice. Forward compat bound added.